drizzlepac/alignimages.py

```diff
@@ -202,361 +197,480 @@ def perform_align(input_list, **kwargs):
-                    if not tweakwcs_info_key.startswith("matched"):
-                        if tweakwcs_info_key.lower() == 'rms':
-                            filteredTable[index]['rms_x'] = item.meta['fit_info'][tweakwcs_info_key][0]
-                            filteredTable[index]['rms_y'] = item.meta['fit_info'][tweakwcs_info_key][1]
-
-                filteredTable[index]['catalog'] = item.meta['fit_info']['catalog']
-                filteredTable[index]['catalogSources'] = len(reference_catalog)
-                filteredTable[index]['matchSources'] = item.meta['fit_info']['nmatches']
-                filteredTable[index]['rms_ra'] = item.meta['fit_info']['RMS_RA'].value
-                filteredTable[index]['rms_dec'] = item.meta['fit_info']['RMS_DEC'].value
-                filteredTable[index]['fit_rms'] = item.meta['fit_info']['FIT_RMS']
-                filteredTable[index]['total_rms'] = item.meta['fit_info']['TOTAL_RMS']
-                filteredTable[index]['offset_x'], filteredTable[index]['offset_y'] = item.meta['fit_info']['shift']
-                filteredTable[index]['scale'] = item.meta['fit_info']['scale'][0]
-                filteredTable[index]['rotation'] = item.meta['fit_info']['rot']
-
-                # populate filteredTable fields "status", "compromised" and
-                # "processMsg" with fitStatusDict fields "valid", "compromised"
-                # and "reason".
-                explicitDictKey ="{},{}".format(item.meta['name'], item.meta['chip'])
-                if fitStatusDict[explicitDictKey]['valid'] == True:
-                    filteredTable[index]['status'] = 0
+            extracted_sources = generate_source_catalogs(process_list,
+                                                         centering_mode='starfind',
+                                                         nlargest=num_sources,
+                                                         output=output)
+
+        for imgname in extracted_sources.keys():
+            table = extracted_sources[imgname]["catalog_table"]
+
+            # Get the location of the current image in the filtered table
+            index = np.where(filtered_table['imageName'] == imgname)[0][0]
+
+            # First ensure sources were found
+
+            if table is None or not table[1]: 
+                log.warning("No sources found in image {}".format(imgname))
+                filtered_table[:]['status'] = 1
+                filtered_table[:]['processMsg'] = "No sources found"
+                current_dt = datetime.datetime.now()
+                delta_dt = (current_dt - starting_dt).total_seconds()
+                log.info('Processing time of [STEP 4]: {} sec'.format(delta_dt))
+                return
+
+            # The catalog of observable sources must have at least MIN_OBSERVABLE_THRESHOLD entries to be useful
+            total_num_sources = 0
+            for chipnum in table.keys():
+                total_num_sources += len(table[chipnum])
```

Comment:
```total_num_sources = sum(map(len, table.values()))``` - this replaces the loop above. I may have commented on this before in a previous PR.

Reply:
`table.values()` does not work. Table object is not a dictionary. Is this table storing *iterables* in a single cell?!

Reply:
I really don't understand what this code is doing. Even for a masked table, `len(table[colname])` returns the total number of rows, not just the unmasked one.

Reply:
Ah, okay... I get it now. You named a dictionary a `table`. I am just gonna step away now.